src/Raven.Client/Documents/Operations/AI/Agents/AiAgentChatReductionConfiguration.cs

```diff
@@ -0,0 +1,218 @@
+        public AiAgentSummarizationByTokens Tokens { get; set; }
+
+        /// <summary>
+        /// Truncates older chat messages when the number of messages exceeds a maximum length.
+        /// Cannot be used together with <see cref="Tokens"/>.
+        /// </summary>
+        public AiAgentTruncateChat Truncate { get; set; }
+
+        /// <summary>
+        /// Configuration settings for storing AI agent conversation history.
+        /// </summary>
+        /// <remarks>
+        /// If this field is <c>null</c>, no history documents will be created or saved
+        /// for the chat when a reduction operation (truncation or summarization) occurs.
+        /// </remarks>
+        public AiAgentHistoryConfiguration History;
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(Tokens)] = Tokens?.ToJson(),
+                [nameof(Truncate)] = Truncate?.ToJson(),
+                [nameof(History)] = History?.ToJson()
+            };
+        }
+    }
+
+    /// <summary>
+    /// Configuration settings for AI agent conversation summarization.
+    /// </summary>
+    /// <remarks>
+    /// Defines how and when the AI agent should summarize the conversation history
+    /// into a concise system prompt to continue the dialogue.
+    /// </remarks>
+    public class AiAgentSummarizationByTokens
+    {
+        private const int DefaultMaxTokensBeforeSummarization = 32 * 1024;
+
+        /// <summary>
+        /// Instruction text prepended to the serialized conversation when requesting a summary.
+        /// </summary>
+        /// <value>
+        /// A detailed directive that provides context, formatting guidance, and any special requirements
+        /// for how the AI should summarize the preceding chat history.
+        /// </value>
+        /// <remarks>
+        /// This prompt is sent with the <c>system</c> role and appears before any summary content.
+        /// Customize it to influence the structure, tone, or depth of the generated summary.
+        /// </remarks>
+        public string SummarizationTaskBeginningPrompt { get; set; }
+
+        /// <summary>
+        /// The user-role message that triggers the summarization process.
+        /// </summary>
+        /// <value>
+        /// A concise reminder instructing the AI to review the entire prior conversation
+        /// and produce a summary according to the guidelines defined in <see cref="SummarizationTaskBeginningPrompt"/>.
+        /// </value>
+        /// <remarks>
+        /// This prompt is sent immediately after <c>SystemPrompt</c> and represents the actual
+        /// request to generate the summary. Adjust its phrasing to change how explicitly
+        /// the model understands the summarization task.
+        /// </remarks>
+        public string SummarizationTaskEndPrompt { get; set; }
+
+        public string ResultPrefix { get; set; } = "Summary of previous conversation: ";
```

Comment:
Should not be initialized here, but same as `SummarizationTaskEndPrompt `

Reply:
Done here:
https://github.com/ravendb/ravendb/pull/20973